File: docs/sphinx/snippets/python/using/examples/hadamard_gate.py
# ============================================================================ #
# Copyright (c) 2022 - 2024 NVIDIA Corporation & Affiliates.                   #
# All rights reserved.                                                         #
#                                                                              #
# This source code and the accompanying materials are made available under     #
# the terms of the Apache License 2.0 which accompanies this distribution.     #
# ============================================================================ #

#[Begin Docs]
import cudaq


@cudaq.kernel
def kernel():
    # A single qubit initialized to the ground/ zero state.
    qubit = cudaq.qubit()

    # Apply Hadamard gate to single qubit to put it in equal superposition.
    h(qubit)

    # Measurement operator.
    mz(qubit)


result = cudaq.sample(kernel, shots_count=1000)

print(result)
#[End Docs]

File: docs/sphinx/snippets/cpp/using/examples/hadamard_gate.cpp
/*******************************************************************************
 * Copyright (c) 2025 NVIDIA Corporation & Affiliates.                         *
 * All rights reserved.                                                        *
 *                                                                             *
 * This source code and the accompanying materials are made available under    *
 * the terms of the Apache License 2.0 which accompanies this distribution.    *
 ******************************************************************************/

// [Begin Docs]
#include <cudaq.h>
#include <iostream>

__qpu__ void hadamard_gate() {
  // Allocate a qubit.
  cudaq::qubit q;
  // Apply a hadamard gate to the qubit.
  h(q);
  // Measure the qubit.
  mz(q);
}

int main() {
  // Call the kernel to run the hadamard gate.
  auto result = cudaq::sample(hadamard_gate);
  std::cout << "Measured |0> with probability "
            << static_cast<double>(result.count("0")) /
                   std::accumulate(result.begin(), result.end(), 0)
            << std::endl;
  std::cout << "Measured |1> with probability "
            << static_cast<double>(result.count("1")) /
                   std::accumulate(result.begin(), result.end(), 0)
            << std::endl;
  return 0;
}
// [End Docs]
